src/apps/reservation-list/list/reservation-list.tsx

```diff
@@ -114,23 +144,166 @@ const ReservationList: FC<ReservationListProps> = ({ pageSize }) => {
+          "identifier",
+          id
+        );
+      }
+      if (
+        reservationFound?.length === 0 &&
+        readyForPickupReservationsPublizon
+      ) {
+        reservationFound = getFromListByKey(
+          [...readyForPickupReservationsPublizon],
+          "identifier",
+          id
+        );
+      }
+      if (reservationFound && reservationFound.length > 0) {
+        return reservationFound[0];
+      }
+      return null;
+    },
+    [
+      readyForPickupReservationsFBS,
+      readyForPickupReservationsPublizon,
+      reservedReservationsFBS,
+      reservedReservationsPublizon
+    ]
+  );
+
+  useEffect(() => {
+    if (modalDetailsId) {
+      const reservationForModal = findReservationInLists(modalDetailsId);
+
+      if (reservationForModal) {
+        setReservation(reservationForModal);
+      }
+    }
+  }, [findReservationInLists, modalDetailsId]);
+
+  useEffect(() => {
+    if (modalDeleteId) {
+      const reservationForModal = findReservationInLists(
+        modalDeleteId.toString()
+      );
+
+      if (reservationForModal) {
+        openReservationDetailsModal(reservationForModal);
+        setReservationToDelete(reservationForModal);
+      }
+    }
+  }, [findReservationInLists, modalDeleteId, openReservationDetailsModal]);
+
+  useEffect(() => {
+    const modalUrlParam = getUrlQueryParam("modal");
+    // if there is a loan details query param, loan details modal should be opened
+    const resDetails = reservationDetails as string;
+    if (modalUrlParam && modalUrlParam.includes(resDetails as string)) {
+      const reservationDetailsModalId = getDetailsModalId(
+        modalUrlParam,
+        resDetails
+      );
+      if (reservationDetailsModalId) {
+        setModalDetailsId(reservationDetailsModalId);
+      }
+    }
+    const deleteRes = deleteReservation as string;
+    if (modalUrlParam && modalUrlParam.includes(deleteRes as string)) {
+      const deleteReservationModalId = getDetailsModalId(
+        modalUrlParam,
+        deleteRes
+      );
+      if (deleteReservationModalId) {
+        setModalDeleteId(deleteReservationModalId);
+      }
+    }
+  }, [deleteReservation, reservationDetails]);
+
   return (
-    <div className="reservation-list-page">
-      <h1 className="text-header-h1 m-32">{t("reservationListHeaderText")}</h1>
-      {user && <ReservationPauseToggler user={user} />}
-      {allListsEmpty && <EmptyReservations />}
-      {!allListsEmpty && (
-        <DisplayedReservations
-          readyForPickupReservationsFBS={readyForPickupReservationsFBS}
-          readyForPickupReservationsPublizon={
-            readyForPickupReservationsPublizon
-          }
-          reservedReservationsFBS={reservedReservationsFBS}
-          reservedReservationsPublizon={reservedReservationsPublizon}
-          pageSize={pageSize}
+    <>
+      <div
+        className="reservation-list-page"
+        style={modalIds.length > 0 ? { display: "none" } : {}}
```

Comment:
I prefer not to render anything if it is not necessary. That helps the performance as well.
If you need to hide it like this I think you could use the class`visually-hidden` instead (I think that is the correct name)

Reply:
I'll not render it.